airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/StateMessageHelper.java

```diff
@@ -0,0 +1,30 @@
+package io.airbyte.protocol.models;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.vavr.control.Either;
+import java.util.List;
+
+public class StateMessageHelper {
+
+  public static class AirbyteStateMessageListTypeReference extends TypeReference<List<AirbyteStateMessage>> {
+  }
+
+  /**
+   * This a takes a json blob state and tries return either a legacy state in the format of a json object or a state message with the new format
+   * which is a list of airbyte state message.
+   * @param state
+   * @return Either a json blob (on the left) or a structure state message.
+   */
+  public static Either<JsonNode, List<AirbyteStateMessage>> getTypeState(JsonNode state) {
+    if (state == null) {
```

Comment:
Aren't we going to always default new workflows to the legacy case even when we should be using the new state format?

Reply:
That's true in this case I will return a either.right(empty()) to show that it is not legacy but not determined.